backend/src/nodes/nodes/image_adjustment/color_levels.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as ImageAdjustmentCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, SliderInput, BoolInput
+from ...properties.outputs import ImageOutput
+from ...impl.image_utils import as_3d
+from ...utils.utils import get_h_w_c
+
+
+@NodeFactory.register("chainner:image:color_levels")
+class ColorLevelsNode(NodeBase):
+    """
+    Color Levels can be used to make an image lighter or darker,
+    to change contrast or to correct a predominant color cast.
+
+    This code was adapted from a Stack-Overflow answer by Iperov,
+    can found at: https://stackoverflow.com/a/60339950
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.description = "Adjust color levels"
+        self.inputs = [
+            ImageInput(),
```

Comment:
This code assumes grayscale, RGB, or RGBA images. Chainner doesn't guarantee that. We have no way to create e.g. 2 channel images right now, but we might add this in the future. 
But there is an easy way to let the `ImageInput` enforce the number of channels of the input image. This will provide the guarantees you need. You can even remove the `c = min(c, 4)` since it's no longer necessary.

```suggestion
            ImageInput(channels=[1, 3, 4]),
```